File: app/src/main/java/nyc/friendlyrobot/demo/ui/reddit/RedditRecyclerView.java
package nyc.friendlyrobot.demo.ui.reddit;

import android.content.Context;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.AttributeSet;

import java.util.List;

import javax.inject.Inject;

import nyc.friendlyrobot.demo.data.model.Post;
import nyc.friendlyrobot.demo.ui.base.BaseActivity;

public class RedditRecyclerView extends RecyclerView implements RedditMVPView {















    private void setLayoutmanager() {
        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
        layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        setLayoutManager(layoutManager);
    }



}

File: app/src/main/java/nyc/friendlyrobot/demo/ui/reddit/RedditRecyclerView.kt
package nyc.friendlyrobot.demo.ui.reddit

import android.content.Context
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.AttributeSet
import nyc.friendlyrobot.demo.ui.base.BaseActivity
import javax.inject.Inject

/**
 * Created by brianplummer on 12/16/15.
 */
class RedditRecyclerView @JvmOverloads constructor(context: Context, attrs: AttributeSet = null!!, defStyle: Int = 0) : RecyclerView(context, attrs, defStyle) {

    @Inject
    lateinit  var postAdapter: PostAdapter

    init {
        (context as BaseActivity).activityComponent.inject(this)
    }

    override fun onFinishInflate() {
        super.onFinishInflate()

        setOrientation()
        adapter = postAdapter
    }

    private fun setOrientation() {
        val layoutManager = LinearLayoutManager(context)
        layoutManager.orientation = LinearLayoutManager.VERTICAL
        setLayoutManager(layoutManager)
    }
}
